crates/cdk-sql-common/src/mint/auth/mod.rs

```diff
@@ -15,36 +16,43 @@ use tracing::instrument;
 use super::{sql_row_to_blind_signature, sql_row_to_keyset_info, SQLTransaction};
 use crate::column_as_string;
 use crate::common::migrate;
-use crate::database::{DatabaseConnector, DatabaseTransaction};
+use crate::database::{ConnectionWithTransaction, DatabaseConnector};
 use crate::mint::Error;
+use crate::pool::{Pool, PooledResource, ResourceManager};
 use crate::stmt::query;
 
 /// Mint SQL Database
 #[derive(Debug, Clone)]
-pub struct SQLMintAuthDatabase<DB>
+pub struct SQLMintAuthDatabase<DB, RM, C, E>
 where
     DB: DatabaseConnector,
+    RM: ResourceManager<Resource = DB, Config = C, Error = E>,
+    C: Debug + Clone + Send + Sync,
+    E: Debug + std::error::Error + Send + Sync + 'static,
```

Comment:
This PR looks good my only concern is the complexity with the generics here. I think it would be best if we could simplify it a bit.